### PR TITLE
Correct signatures.

### PR DESCRIPTION
**Pull request by nojaf**
In light of my investigation of the determinism of https://github.com/dotnet/fsharp/pull/14494, I found out that the `mvid` is stable (when using graph type checking) with these tweaks.

For `TypedTree.fsi` I'm just updating the signature with what is actually used. It is a bit weird that these were different in the first place.

For `local.fsi`, the `_` leads to using the internal name 
image, which again I don't think should have been there in the first place.

I believe the remaining determinism problem is with the `pdb` generation. But that is still a hunch at this point.

### CODE REVIEW

File: src/FSharp.Core/local.fsi [hunk 1]
```diff
@@ -6,22 +6,22 @@ open Microsoft.FSharp.Core
 
 [<AutoOpen>]
 module internal DetailedExceptions =
-    val inline invalidArgFmt: arg: string -> format: string -> paramArray: obj array -> _
```
Comment (abonie) [inline src/FSharp.Core/local.fsi:9]: So will using underscore like here, break determinism together with graph based tc?

Comment (nojaf) [reply src/FSharp.Core/local.fsi:9]: Yes, because the signature will produce a name with a generated id (see screenshot).

**Comment by vzarytovskii** [comment]
So, this is the fix for the FCS itself, right? Do you have any suggestions how can we fix it for everyone without turning off the graph checking? 
We also should have a separate issue for investigating PDB issue, I thinkg.

File: src/FSharp.Core/local.fsi (hunk 1 above)
Comment (abonie) [reply src/FSharp.Core/local.fsi:9]: Right, so we should probably have at least a tracking issue to either: solve this by making this generation deterministic (regardless of compilation order) or produce some sort of warning from graph based tc when this construct is used?

**Review by T-Gro — APPROVED** [review]

**Review by abonie — APPROVED** [review]

**Comment by auduchinok** [comment]
Would it be possible to keep these signatures as is, but to fix the logic that causes the issues? Or at least we should later add tests for these cases, so they assert that they still work with determinism enabled.

**Comment by nojaf** [comment]
> Would it be possible to keep these signatures as is, but to fix the logic that causes the issues?

In theory yes, but I'm unable to reproduce this problem outside of `FCS` or `FSharp.Core`.
On my local machine, I created a project where I have both problems in 50 different file pairs.
These all have no relation and are processed in parallel. After compiling 50 times, the `mvid` is exactly the same.

At this point, I'd rather have this merged in and wait for a more obvious case to appear in the wild.

File: src/FSharp.Core/local.fsi (hunk 1 above)
Comment (safesparrow) [reply src/FSharp.Core/local.fsi:9]: I think this construct is common-enough that we need to solve it anyway.

**Comment by safesparrow** [comment]
>At this point, I'd rather have this merged in and wait for a more obvious case to appear in the wild.

Regardless of whether it's merged or not, we should use the case we have at hands as it can help us solve the general problem.
If we do merge it, we can always use a historical revision of the codebase for determinism testing, but that makes it slightly more complicated for an average contributor.

**Pull request merged**